content/departments/product-engineering/engineering/code-graph/search/guiding_principles.md

```diff
@@ -0,0 +1,41 @@
+# Guiding principles for teamwork on Search Product
+
+This document outlines some guiding principles we seek to follow when welcoming engineers onto the Search Product team. It also serves as shared vocabulary on the team, when we talk about the culture we try to foster.
+
+## You're trusted by default
+
+The notion that one must "earn the trust" of their peers or manager is extremely limiting for newcomers, who are faced with this arbitrary "bar of trust" they must meet before they can deploy agency.
+
+On this team, trust is a given. Your teammates will value the opinions, questions, ideas and concerns you will bring forward from day zero. When we ask you questions, it's out of genuine interest for what you might have to answer, not as a test of your worth, and your answers will not be judged.
+
+## Liberating ability through context
+
+We trust in your skills and ability; this is why you are on this team. But raw ability must be coupled with context on the team's technical domain and its goals in order to produce impact.
+
+Our job as you onboard is to give you this context, so that you can liberate your ability. As a result, your job as you onboard is to focus on absorbing context, rather than attempting to demonstrate ability through immediate execution. This doesn't mean operating solely in the abstract, as of course part of absorbing context will entail tackling some practical problems.
+
+## We'll provide direction, not tasks
+
+While we will provide you with some initial tasks that will help you absorb context, the eventual end goal is that you are able to identify the work you should tackle to produce impact towards team and company goals. This impact can be direct or indirect. An example of direct impact is building functionality that will directly unblock a user outcome. An example of indirect impact is ensuring we have a solid foundation to build upon, so that we can build functionality that unblocks user outcomes.
```

Comment:
I do like your rewording regardless of the "absorb context" topic. Thanks :)

```suggestion
## We'll provide direction, not tasks

While we will provide you with some initial tasks to help you understand how we work, the eventual end goal is that you are able to identify the work you should tackle to produce impact towards team and company goals. This impact can be direct or indirect. An example of direct impact is building functionality that will directly unblock a user outcome. An example of indirect impact is ensuring we have a solid foundation to build upon, so that we can build functionality that unblocks user outcomes.
```